test/extended/util/framework.go

```diff
@@ -246,6 +247,170 @@ func VarSubOnFile(srcFile, destFile, varToSub, val string) error {
+	return stdout, stderr, err
+}
+
+var buildPathPattern = regexp.MustCompile(`^build/([\w\-\._]+)$`)
+
+type BuildResult struct {
+	// BuildPath is a resource qualified name (e.g. "build/test-1").
+	BuildPath string
+	// StartBuildStdErr is the StdErr output generated by oc start-build.
+	StartBuildStdErr string
+	// StartBuildErr is the error, if any, returned by the direct invocation of the start-build command.
+	StartBuildErr error
+	// The buildconfig which generated this build.
+	BuildConfigName string
+	// Build is the resource created. May be nil if there was a timeout.
+	Build *buildapi.Build
+	// BuildAttempt represents that a Build resource was created.
+	// false indicates a severe error unrelated to Build success or failure.
+	BuildAttempt bool
+	// BuildSuccess is true if the build was finshed successfully.
+	BuildSuccess bool
+	// BuildFailure is true if the build was finished with an error.
+	BuildFailure bool
+	// BuildTimeout is true if there was a timeout waiting for the build to finish.
+	BuildTimeout bool
+	// The openshift client which created this build.
+	oc *CLI
+}
+
+// DumpLogs sends logs associated with this BuildResult to the GinkgoWriter.
+func (t *BuildResult) DumpLogs() {
+	fmt.Fprintf(g.GinkgoWriter, "\n\n*****************************************\n")
+	fmt.Fprintf(g.GinkgoWriter, "Dumping Build Result: %#v\n", *t)
+
+	if t == nil {
+		fmt.Fprintf(g.GinkgoWriter, "No build result available!\n\n")
+		return
+	}
+
+	desc, err := t.oc.Run("describe").Args(t.BuildPath).Output()
+
+	fmt.Fprintf(g.GinkgoWriter, "\n** Build Description:\n")
+	if err != nil {
+		fmt.Fprintf(g.GinkgoWriter, "Error during description retrieval: %+v\n", err)
+	} else {
+		fmt.Fprintf(g.GinkgoWriter, "%s\n", desc)
+	}
+
+	fmt.Fprintf(g.GinkgoWriter, "\n** Build Logs:\n")
+
+	bldOuput, err := t.Logs()
+	if err != nil {
+		fmt.Fprintf(g.GinkgoWriter, "Error during log retrieval: %+v\n", err)
+	} else {
+		fmt.Fprintf(g.GinkgoWriter, "%s\n", bldOuput)
+	}
+
+	fmt.Fprintf(g.GinkgoWriter, "\n\n")
+
+	// if we suspect that we are filling up the registry file system, call ExamineDiskUsage / ExaminePodDiskUsage
+	// also see if manipulations of the quota around /mnt/openshift-xfs-vol-dir exist in the extended test set up scripts
+	ExamineDiskUsage()
+	ExaminePodDiskUsage(t.oc)
+
+	fmt.Fprintf(g.GinkgoWriter, "\n\n")
+}
+
+// Logs returns the logs associated with this build.
+func (t *BuildResult) Logs() (string, error) {
+	if t == nil || t.BuildPath == "" {
+		return "", fmt.Errorf("Not enough information to retrieve logs for %#v", *t)
+	}
+
+	bldOuput, err := t.oc.Run("logs").Args("-f", t.BuildPath).Output()
+	if err != nil {
+		return "", fmt.Errorf("Error retieving logs for %#v: %v", *t, err)
+	}
+
+	return bldOuput, nil
+}
+
+// Dumps logs and triggers a Ginkgo assertion if the build did NOT succeed.
+func (t *BuildResult) AssertSuccess() *BuildResult {
+	if !t.BuildSuccess {
+		t.DumpLogs()
+	}
+	o.Expect(t.BuildSuccess).To(o.BeTrue())
```

Comment:
throwing gomega errors from utility functions is problematic because the test report output will report this line as the point of failure, instead of the actual area in the test.  We tend to try to avoid doing this.